test/components/date-picker.spec.tsx

```diff
@@ -167,6 +167,114 @@ describe('The DatePicker Component', () => {
         await waitForDom(() => expect(datePicker.selectedDate).to.equal('Sun Jan 01 2017'));
     });
 
+    it('should set the tabindex attribute', () => {
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker tabIndex={123} />)
+            .withDriver(DatePickerTestDriver);
+
+        return waitForDom(() => expect(datePicker.root).to.have.attribute('tabindex', '123'));
+    });
+
+    it('should set the required attribute', () => {
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker required />)
+            .withDriver(DatePickerTestDriver);
+
+        return waitForDom(() => expect(datePicker.input).to.have.attribute('required'));
+    });
+
+    it('should set the name attribute', () => {
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker name="test" />)
+            .withDriver(DatePickerTestDriver);
+
+        return waitForDom(() => expect(datePicker.input).to.have.attribute('name', 'test'));
+    });
+
+    it('should focus automatically on the datepicker when autoFocus is provided', async () => {
+        const {driver: datePicker} = clientRenderer.render(<DatePicker autoFocus />)
+            .withDriver(DatePickerTestDriver);
+
+        await waitFor(() => expect(document.activeElement).to.equal(datePicker.input));
+    });
+
+    describe('When Disabled or ReadOnly', () => {
+        it('should not show the dropdown when the icon is clicked', async () => {
+            const {driver: datePicker, waitForDom, container} = clientRenderer.render(<DatePicker disabled />)
+                .withDriver(DatePickerTestDriver);
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            datePicker.clickOnDatePicker();
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            clientRenderer.render(<DatePicker readOnly />, container);
```

Comment:
When you are rendering here, you don't use with driver. I don't know if this is a problem or not but below you are using it.